src/Microsoft.Health.Api/Features/ByteCounter/ByteCountingStream.cs

```diff
@@ -0,0 +1,135 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Api.Features.ByteCounter
+{
+    /// <summary>
+    /// A stream to wrap an underlying stream and counts the number of bytes passed while operating with it.
+    /// </summary>
+    public class ByteCountingStream : Stream
```

Comment:
Perhaps a name more like `MonitoredStream` or `AuditedStream` to mirror `BufferedStream`?

Reply:
I think AuditedStream could get too easily confused with the Audit logging / audit feature. But I like the name MonitoredStream